clients/fossid-webapp/src/main/kotlin/model/result/MatchedLines.kt

```diff
@@ -19,7 +19,9 @@
 
 package org.ossreviewtoolkit.clients.fossid.model.result
 
+import org.ossreviewtoolkit.clients.fossid.PolymorphicList
+
 data class MatchedLines(
-    val localFile: Map<String, String>,
-    val mirrorFile: Map<String, String>
+    val localFile: PolymorphicList<Int>,
```

Comment:
I didn't ask them (and I would prefer not too since I don't want to harass my contact 😄) however I really believe this is the case and it's a bug / bad design / technical quirk on their side.

As seen here (https://github.com/oss-review-toolkit/ort/blob/152348f7d2d87e4aeac322b69e01906e1cb9bbd5/clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt#L114),  FossID sometimes returns some strange answers.

For instance **files_and_folders/get_fossid_results** returns:
```
"data": {
		"241": {
			"id": "241",
			"created": "2023-05-25 17:22:11",
			"scan_id": "0",
			"scan_file_id": "25436135",
```
You can notice that there is not point to return the results as a map with the entity id as a key, since the id is also present in the object: an array would be enough.
In a way, the matched line response is the same, the object returned is aggregated by its line number, but this is even weirder because the object _IS_ just a line number.

Another example to showcase such quirk. The response of **projects/get_all_scans**:

```
{
	"operation": "projects_get_all_scans",
	"status": "1",
	"data": [
		{
			"id": "8911",
			"0": "8911",
			"created": "2023-03-01 12:55:53",
			"1": "2023-03-01 12:55:53",
			"updated": "2023-03-01 12:56:04",
			"2": "2023-03-01 12:56:04",
```
Here it seems to be a badly configured serializer.